SCClassLibrary/Common/Math/SimpleNumber.sc

```diff
@@ -705,13 +711,13 @@ SimpleNumber : Number {
 		};
 		hours = (decimal.div(3600) % 24).asString.padLeft(2, "0").add($:);
 		minutes = (decimal.div(60) % 60).asString.padLeft(2, "0").add($:);
-		seconds = (decimal % 60).asString.padLeft(2, "0").add($.);
-
-		// min value of precision is 0.001; this ensures that we stick to 3 decimal places in the
-		// formatted string.
-		precision = max(precision, 0.001);
-		mseconds = this.frac.round(precision) * 1000;
-		mseconds = mseconds.round.asInteger.asString.padLeft(3, "0");
+		seconds = (decimal % 60).asString.padLeft(2, "0");
+		if(decimalPlaces > 0, {
+			// this could be simplified once sclang gains sprintf functionality, see issue #3570
+			mseconds = "." ++ (number.frac.round(precision) * pow(10, decimalPlaces)).asInteger.asString.padLeft(decimalPlaces, "0");
+		}, {
+			mseconds = "";
+		});
```

Comment:
i think i left a note about this on another PR, generally we write this with trailing closure syntax. also, you can assign the result of `if` to an object, so i think this would be more clearly written as:

```supercollider
mseconds = if(decimalPlaces > 0) { "." ++ ... } { "" };
```